tasks/deploy/github.py

```diff
@@ -0,0 +1,157 @@
+
+errno_regex = re.compile(r".*\[Errno (\d+)\] (.*)")
+
+__all__ = ["Github", "GithubException"]
+
+
+class GithubException(Exception):
+    pass
+
+
+class Github(object):
+    BASE_URL = "https://api.github.com"
+
+    def __init__(self, api_token=None):
+        self.api_token = api_token if api_token else self._api_token()
+
+    def test_repo_found(self, repo):
+        """
+        Checks if a repo can be found. This is useful for testing access permissions to repos.
+        """
+        result = self.repo(repo)
+
+        # name is arbitrary, just need to check if something is in the result
+        if "name" in result:
+            return
+
+        print("Cannot find Github repo {}".format(repo))
+        print("If you cannot see it in the Github UI, you likely need permission.")
+        raise Exit(code=1)
+
+    def repo(self, repo_name):
+        """
+        Gets the repo info.
+        """
+
+        path = "/repos/{}".format(repo_name)
+        return self.make_request(path, json=True)
+
+    def trigger_workflow(self, repo_name, workflow_name, ref, inputs=None):
+        """
+        Create a pipeline targeting a given reference of a project.
+        ref must be a branch or a tag.
+        """
+        if inputs is None:
+            inputs = {}
+
+        path = "/repos/{}/actions/workflows/{}/dispatches".format(repo_name, workflow_name)
+        data = json.dumps({"ref": ref, "inputs": inputs})
+        return self.make_request(path, data=data)
+
+    def workflow_run(self, repo_name, run_id):
+        """
+        Gets info on a specific workflow.
+        """
+        path = "/repos/{}/actions/runs/{}".format(repo_name, run_id)
+        return self.make_request(path, json=True)
+
+    def download_artifact(self, repo_name, artifact_id, destination_dir):
+        """
+        Downloads the artifact identified by artifact_id to destination_dir.
+        """
+        path = "/repos/{}/actions/artifacts/{}/zip".format(repo_name, artifact_id)
+        content = self.make_request(path, raw_content=True)
+
+        zip_target_path = os.path.join(destination_dir, "{}.zip".format(artifact_id))
+        with open(zip_target_path, "wb") as f:
+            f.write(content)
+        return zip_target_path
+
+    def workflow_run_artifacts(self, repo_name, run_id):
+        """
+        Gets list of artifacts for a workflow run.
+        """
+        path = "/repos/{}/actions/runs/{}/artifacts".format(repo_name, run_id)
+        return self.make_request(path, json=True)
+
+    def latest_workflow_run_for_ref(self, repo_name, workflow_name, ref):
+        """
+        Gets latest workflow run for a given reference
+        """
+        runs = self.workflow_runs(repo_name, workflow_name)
+        ref_runs = [run for run in runs["workflow_runs"] if run["head_branch"] == ref]
+
+        if len(ref_runs) == 0:
+            return None
+
+        return sorted(ref_runs, key=lambda run: run['created_at'], reverse=True)[0]
```

Comment:
I think this is equivalent and simpler
```suggestion
        return max(ref_runs, key=lambda run: run['created_at'], default=None)
```
but please double check.